### PR TITLE
Updated the Socios edit form to make it look as the design

### LINKED ISSUE #259: Socios form is not working

### LINKED ISSUE #261: Add styles to edit Member form
### Explication

I checked and this form doesn't have the styles of the design

### Screenshots

imagen

### PR DESCRIPTION
**Pull request by heliomar-pena**
### Description

I updated the Socios edit form, so it look as the design. I had to change the mutation to edit the member, since they was split in two forms but on the design was just 1 form.

So now the mutation is changing the user and the member at the same time, but you needs to pass the userId and the memberId, you could simply create a input type hidden with this data, and you'll receive that in the submitForm function

The rest of the variables in the form match the variables expected to receive the mutation.

This PR is DRAFT since I couldn't finish of make it works

### Update

Hi, im @Eghinner

Create a PUT method endpoint to handle the form request. Also, its respective logic in the controller takes the changes that victor made in the mutation.

Related to the above, I made a change in the types to accept `memberId` and use the mutation. I also set the default value of `memberId` in the form.

### Disclaimer

There are some users that do not have member data. I did not do the validation for those cases

It would be to delete those users in the database or create an issue for that validation.

fixes #261 
fixes #259 

#### Screenshots

imagen

#### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How to test it

If applicable, describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

1. Go to deploy
2. Enter as admin
3. Go to Socios
4. click on edit some socio

### Checklist:

- [x] Doesn't break the current code.
- [x] Passes linters and test, also is building.
- [x] Doesn't have spelling or grammatical problems.
- [x] Doesn't have unnecessary comments or debugging code.
- [x] The branch is updated with the last dev branch changes.

### CODE REVIEW

**Comment by angelik0828** [comment]
## :heavy_check_mark:  Implemented design
image

**Review by Israel-Laguan — APPROVED** [review]

File: pages/socios/editar/[id].tsx [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
+  const [loading, setLoading] = useState(false)
   const { error: logError } = useLogger()
 
-  const submitUserHandler = async (updatedUser: UserEditable) => {
-    await updateUser({
-      variables: { ...updatedUser, id: query.id },
-    })
-    push('/socios')
+  const submitMemberHandler = async (user: UserEditable) => {
+    console.log(user)
```
Comment (Israel-Laguan) [inline pages/socios/editar/[id].tsx]: remove

File: pages/socios/editar/[id].tsx [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
-    )
-
   return (
-    <EditUserAndMember
-      submitUser={submitUserHandler}
-      submitMember={submitMemberHandler}
-      updateUserMutationLoading={updateUserMutationLoading}
-      updaterMemberMutationLoading={updaterMemberMutationLoading}
-    />
+    <div>
```
Comment (Israel-Laguan) [inline pages/socios/editar/[id].tsx]: Maybe this div is not needed

File: services/GraphQL/users/types.d.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ export interface MemberBase {
 
 export interface MemberEditable extends MemberBase {
   email: string
+  memberCode?: string
+  memberId?: string
```
Comment (Israel-Laguan) [inline services/GraphQL/users/types.d.ts]: Can we try to make this one NOT nullable?

File: services/GraphQL/users/types.d.ts [hunk 4]
```diff
@@ -54,6 +58,7 @@ export const usersInfo = gql`
     position
     memberCode: member_code
     avatarUrl: avatar_url
+    password: password
```
Comment (Israel-Laguan) [inline services/GraphQL/users/types.d.ts]:
```suggestion
    password
```

File: views/Users/Edit/EditMember/index.tsx [hunk 5]
```diff
@@ -1,35 +1,109 @@
...
+              {...register('password', { required: true })}
+            />
+            {errors.password && <small className={styles['error-message']}>{errors.password.message}</small>}
+          </div>
+        </div>
+      </section>
+      <section className={btnStyle['buttons-container']}>
+        <Button iconName="" className={btnStyle['button-save']} type="submit">
+          {buttonText}
+        </Button>
```
Comment (Israel-Laguan) [inline views/Users/Edit/EditMember/index.tsx:106]: Maybe add a "Cancelar" button like other forms

**Review by Israel-Laguan — APPROVED** [review]

**Comment by Israel-Laguan** [comment]
@angelik0828 please check one more time

**Comment by angelik0828** [comment]
## ✔️ Has been verified and found

- When editing, the data of the existing member is loaded
- Partner Update Confirmation

image

image


##  ⚠️ Updating the member's membership date does not update the list

image

**Comment by Eghinner** [comment]
> Updating the member's membership date does not update the list

This is because there is no column in the database for this specific field.

@e-azocar What should we do? `events` has a `date` column.

**Pull request merged**